crates/common/tedge_utils/src/file.rs

```diff
@@ -0,0 +1,112 @@
+use nix::unistd::*;
+use std::fs::File;
+use std::os::linux::fs::MetadataExt;
+use std::os::unix::fs::PermissionsExt;
+use std::{fs, io};
+use users::{get_group_by_name, get_user_by_name};
+
+pub fn create_directory_with_user_group(
+    grp_user: &str,
+    dirs: Vec<&str>,
+) -> Result<(), anyhow::Error> {
+    for directory in dirs {
+        match fs::create_dir(directory) {
+            Ok(_) => {
+                change_owner_and_permission(directory, grp_user, 0o775)?;
+            }
+
+            Err(e) => {
+                if e.kind() == io::ErrorKind::AlreadyExists {
+                    return Ok(());
+                } else {
+                    eprintln!(
+                        "failed to create the directory {} due to error {}",
+                        directory, e
+                    );
+                    return Err(e.into());
+                }
+            }
+        }
+    }
+    Ok(())
+}
+
+pub fn create_file_with_user_group(grp_user: &str, files: Vec<&str>) -> Result<(), anyhow::Error> {
```

Comment:
Same comments:

* Focus on a single file.
* Add the mod as an argument
* No `eprintln`

Reply:
Also consider to have two parameters: one for the user name, another one for the group name.

Reply:
fixed